libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFSalesforceAction.m

```diff
@@ -234,4 +259,21 @@ - (BOOL)isEqualToAction:(CSFAction *)action {
     return YES;
 }
 
+#pragma mark SFAuthenticationManagerDelegate
+
+- (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
+    SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
+    if ([accountManager isKindOfClass:[SFUserAccountManager class]]) {
+        if (accountManager.currentUserIdentity.userId != self.enqueuedNetwork.account.credentials.userId) {
```

Comment:
This is the check that should be changed to:

        if (![accountManager.currentUserIdentity isEqual:self.enqueuedNetwork.account.accountIdentity])